hbar_transfer.go

```diff
@@ -0,0 +1,74 @@
+package hedera
+
+import "github.com/hashgraph/hedera-protobufs-go/services"
+
+type HbarTransfer struct {
+	AccountID  *AccountID
+	Amount     Hbar
+	IsApproved bool
+}
+
+type _HbarTransfers struct {
+	transfers []*HbarTransfer
+}
+
+func _HbarTransferFromProtobuf(pb []*services.AccountAmount) []*HbarTransfer {
+	result := make([]*HbarTransfer, 0)
+	for _, acc := range pb {
+		result = append(result, &HbarTransfer{
+			AccountID:  _AccountIDFromProtobuf(acc.AccountID),
+			Amount:     HbarFromTinybar(acc.Amount),
+			IsApproved: acc.GetIsApproval(),
+		})
+	}
+
+	return result
+}
+
+func (transfer *HbarTransfer) _ToProtobuf() *services.AccountAmount { //nolint
+	var account *services.AccountID
+	if transfer.AccountID != nil {
+		account = transfer.AccountID._ToProtobuf()
+	}
+
+	return &services.AccountAmount{
+		AccountID:  account,
+		Amount:     transfer.Amount.AsTinybar(),
+		IsApproval: transfer.IsApproved,
+	}
+}
+
+func (transfers *_HbarTransfers) Len() int {
+	return len(transfers.transfers)
+}
+func (transfers *_HbarTransfers) Swap(i, j int) {
+	transfers.transfers[i], transfers.transfers[j] = transfers.transfers[j], transfers.transfers[i]
+}
+
+func (transfers *_HbarTransfers) Less(i, j int) bool {
+	if transfers.transfers[i].AccountID.Shard < transfers.transfers[j].AccountID.Shard { //nolint
```

Comment:
Since the comparison of `AccountID` has been done several times here (and I think more times elsewhere) we should really just create a `func (a *AccountID) _Compare(b *AccountID) int { ... }`

Reply:
We actually had a `Comapre()` for AccountID and TokenID, no idea why I didn't use it before.